### PR TITLE
Add PowerVR device validator

### PR DESCRIPTION
**Pull request by shaunh-imgtech**
Adds a PowerVR device specific validator (largely based on existing vendor validators) to AGI which validates counters found on PowerVR Rogue, Series A/B/C and Furian based GPUs. Also includes a fallback that runs a more relaxed counter check (similar to the generic validator added in #1247) for new devices that AGI may encounter but not explicitly check for yet.

### CODE REVIEW

**Comment by shaunh-imgtech** [comment]
As discussed over email with Serdar there is a chance that device validation can fail due to render stage events triggered by system processes such as surfaceflinger (which are still using GLES) being intermixed with render stage events from the Vulkan app being executed during device validation, which results in only some render stage events to containing a valid Vulkan command buffer handle. See https://github.com/google/agi/blob/master/gapis/trace/android/validate/validate.go#L241.

**Review by ttanatb — APPROVED** [review]
Unfortunate that there's so much code repetition, but that's a bigger problem for this PR

File: gapis/service/service.proto [hunk 1]
```diff
@@ -1508,6 +1508,11 @@ message DeviceValidationResult {
     // at least one value is non-zero. This is consistent with the CTS test in
     // Android U.
     GENERIC = 3;
+
+    // PowerVR validator checks for specific counters to ensure that the name
+    // and id is consistent with hardcoded values. Also has specific check to
+    // make sure that the value is within expected ranges.
+    POWERVR = 4;
```
[the remark this answers is not in the document]
Comment (shaunh-imgtech) [reply gapis/service/service.proto]: Done

File: gapis/trace/android/powervr/profiling_data.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+	}
+
+	data.Slices.MapIdentifiers(ctx, handleMapping)
+
+	for i := range data.Slices {
+		slice := &data.Slices[i]
+		subOrder, ok := submissionOrdering[slice.Submission]
+		if ok {
+			name := strings.ToLower(slice.Name)
+			if name == "vertex" || name == "fragment" {
```
[the remark this answers is not in the document]
Comment (shaunh-imgtech) [reply gapis/trace/android/powervr/profiling_data.go]: Makes sense, done :)

**Review by yalcinmelihyasin — APPROVED** [review]

**Comment by shaunh-imgtech** [comment]
Hi Guys, I was wondering if we could get this PR merged in?

**Comment by ttanatb** [comment]
Hey sorry for the delay, #1262 just got merged, so you could try pulling from master and see if you're still running into failures from render stage events triggered by system processes.

If all works out, feel free to merge the PR.

**Comment by shaunh-imgtech** [comment]
> Hey sorry for the delay, #1262 just got merged, so you could try pulling from master and see if you're still running into failures from render stage events triggered by system processes.

No worries, I've updated my branch/PR and tested the new validation stuff locally, all looks good on my end, I no longer get the failures.

> If all works out, feel free to merge the PR.

Not sure if this was for me but I don't have permissions to merge :)

**Comment by ttanatb** [comment]
>Not sure if this was for me but I don't have permissions to merge :)

My apologies, thought that approval was enough -- I'll merge once test turns green

**Pull request merged**